### PR TITLE
Support additional CI variables

### PR DESCRIPTION
**Pull request by hverlin**
Fix https://github.com/firecow/gitlab-ci-local/issues/653, https://github.com/firecow/gitlab-ci-local/issues/1112

### CODE REVIEW

File: src/predefined-variables.ts [hunk 1]
```diff
@@ -25,6 +26,9 @@ export function init ({gitData, argv}: PredefinedVariablesOpts): {[name: string]
         CI_COMMIT_BRANCH: gitData.commit.REF_NAME, // Not available in merge request or tag pipelines
         CI_COMMIT_REF_NAME: gitData.commit.REF_NAME, // Tag or branch name
         CI_COMMIT_REF_SLUG: gitData.commit.REF_NAME.replace(/[^a-z\d]+/ig, "-").replace(/^-/, "").slice(0, 63).replace(/-$/, "").toLowerCase(),
+        CI_COMMIT_TIMESTAMP: gitData.commit.TIMESTAMP,
```
Comment (ANGkeith) [inline src/predefined-variables.ts:29]: https://github.com/firecow/gitlab-ci-local/blob/56319b8d8df4f1227653817a900f97bf31bd309c/src/git-data.ts#L33

Do you need a proper timestamp ?  `CI_COMMIT_TIMESTAMP` currently, it's kinda "mock"

Comment (ANGkeith) [reply src/predefined-variables.ts:29]: Depends on how we we look at it, i think the current timestamp actually can make sense too. lets leave it haha

File: src/job.ts [hunk 2]
```diff
@@ -428,6 +431,7 @@ export class Job {
 
         const argv = this.argv;
         this._startTime = process.hrtime();
+        this._variables["CI_JOB_STARTED_AT"] = new Date().toISOString();
```
Comment (ANGkeith) [inline src/job.ts:434]: new Date().toISOString();

actual:
`2024-06-26T16:19:20.644Z`

expected:
`2024-06-26T16:19:20Z`

minor tweak

File: src/predefined-variables.ts [hunk 3]
```diff
@@ -8,6 +8,7 @@ type PredefinedVariablesOpts = {
 
 export function init ({gitData, argv}: PredefinedVariablesOpts): {[name: string]: string} {
     const predefinedVariables: {[key: string]: string} = {
+        CI: "true",
```
[the remark this answers is not in the document]
Comment (hverlin) [reply src/predefined-variables.ts:11]: This one is different, right?
https://github.com/firecow/gitlab-ci-local?tab=readme-ov-file#local-only

`process.env.CI` is checked in many places when scripts run on CI, so I think it should be `true`? 
https://sourcegraph.com/search?q=context:global+process.env.CI&patternType=keyword&sm=0

[the remark this answers is not in the document]
Comment (ANGkeith) [reply src/predefined-variables.ts:11]: agree

**Review by ANGkeith — APPROVED** [review]

File: src/predefined-variables.ts (hunk 1 above)
Comment (hverlin) [reply src/predefined-variables.ts:29]: I am not sure as well :sweat_smile: but given that it's possible to override it, maybe it's better to have at least something by default

**Review by firecow — APPROVED** [review]

**Comment by hverlin** [comment]
@firecow Do you know when you are going to create a new release?

**Comment by firecow** [comment]
@hverlin I'll make one now

**Comment by hverlin** [comment]
Thanks!

**Pull request merged**